common/src/main/java/org/apache/uniffle/common/ShuffleDataResult.java

```diff
@@ -54,28 +61,32 @@ public ShuffleDataResult(byte[] data, List<BufferSegment> bufferSegments) {
   }
 
   public byte[] getData() {
-    if (data == null) {
+    if (buffer == null) {
       return null;
     }
-    if (data.hasArray()) {
-      return data.array();
+    if (buffer.nioByteBuffer().hasArray()) {
+      return buffer.nioByteBuffer().array();
     }
-    return ByteBufUtils.readBytes(data);
+    return ByteBufUtils.readBytes(Unpooled.wrappedBuffer(buffer.nioByteBuffer()));
```

Comment:
Maybe we can `return ByteBufUtils.readBytes(buffer.byteBuf());`

Reply:
ok,I' ll fix it

Reply:
@xumanbu remind to fix this.

Reply:
ok